apprise/plugins/dot.py

```diff
@@ -0,0 +1,636 @@
+#   - ditherKernel (string, optional, default FLOYD_STEINBERG):
+#     dithering kernel.
+
+from contextlib import suppress
+import json
+
+import requests
+
+from ..common import NotifyImageSize, NotifyType
+from ..locale import gettext_lazy as _
+from ..url import PrivacyMode
+from ..utils.parse import parse_bool
+from .base import NotifyBase
+
+# Supported Dither Types
+DOT_DITHER_TYPES = (
+    "DIFFUSION",
+    "ORDERED",
+    "NONE",
+)
+
+# Supported Dither Kernels
+DOT_DITHER_KERNELS = (
+    "THRESHOLD",
+    "ATKINSON",
+    "BURKES",
+    "FLOYD_STEINBERG",
+    "SIERRA2",
+    "STUCKI",
+    "JARVIS_JUDICE_NINKE",
+    "DIFFUSION_ROW",
+    "DIFFUSION_COLUMN",
+    "DIFFUSION_2D",
+)
+
+
+class NotifyDot(NotifyBase):
+    """A wrapper for Dot. Notifications."""
+
+    # The default descriptive name associated with the Notification
+    service_name = "Dot."
+    # Alias: devices marketed as "Quote/0" remain discoverable.
+
+    # The services URL
+    service_url = "https://dot.mindreset.tech"
+
+    # The default protocol
+    protocol = "dot"
+
+    # The default secure protocol
+    secure_protocol = "dot"
+
+    # A URL that takes you to the setup/help of the specific protocol
+    setup_url = "https://dot.mindreset.tech/docs/service/studio/api/text_api"
+
+    # Allows the user to specify the NotifyImageSize object
+    image_size = NotifyImageSize.XY_128
+
+    # Supported API modes
+    SUPPORTED_MODES = ("text", "image")
+    DEFAULT_MODE = "text"
+
+    # Define object templates
+    templates = (
+        "{schema}://{apikey}@{device_id}/{mode}",
+        "{schema}://{apikey}@{device_id}/{mode}/",
+    )
+
+    # Define our template arguments
+    template_tokens = dict(
+        NotifyBase.template_tokens,
+        **{
+            "apikey": {
+                "name": _("API Key"),
+                "type": "string",
+                "required": True,
+                "private": True,
+            },
+            "device_id": {
+                "name": _("Device Serial Number"),
+                "type": "string",
+                "required": True,
+                "map_to": "device_id",
+            },
+            "mode": {
+                "name": _("API Mode"),
+                "type": "choice:string",
+                "values": SUPPORTED_MODES,
+                "default": DEFAULT_MODE,
+                "map_to": "mode",
+            },
+        },
+    )
+
+    # Define our template arguments
+    template_args = dict(
+        NotifyBase.template_args,
+        **{
+            "refresh": {
+                "name": _("Refresh Now"),
+                "type": "bool",
+                "default": True,
+                "map_to": "refresh_now",
+            },
+            "title": {
+                "name": _("Text Title"),
+                "type": "string",
+            },
+            "message": {
+                "name": _("Text Content"),
+                "type": "string",
+            },
```

Comment:
Not sure if I get this. I think I can follow the default title (fallback if one isn't specified), but the body is usually mandatory